src/Microsoft.DotNet.Build.Tasks/Targets/CodeCoverage.targets

```diff
@@ -22,7 +22,8 @@
     <CoverageEnabledForProject Condition="'$(CoverageEnabledForProject)'=='' and '$(IsTestProject)'=='true'">$(CodeCoverageEnabled)</CoverageEnabledForProject>
 
     <GenerateIndividualCoverageReport Condition="'$(BuildAllProjects)'!='true' and '$(CoverageEnabledForProject)'=='true'">true</GenerateIndividualCoverageReport>
-    <GenerateFullCoverageReport Condition="'$(GenerateCodeCoverageReportForAll)'=='true' and '$(CodeCoverageEnabled)'=='true'">true</GenerateFullCoverageReport>
+    <GenerateFullCoverageReport Condition="'$(GenerateCodeCoverageReportForAll)'=='true' and '$(CodeCoverageEnabled)'=='true' and '$(SkipFullCoverageReport)' != 'true'">true</GenerateFullCoverageReport>
```

Comment:
Where do you plan to set SkipFullCoverageReport and GenerateVisitedReport? I was expecting the same property to be used one in the positive and one in the negative case. If you expect these to be mutually exclusive then I would use the same property but if you want potentially do both in some case then having the two properties make sense.